### PR TITLE
fix(TDI-38601): Implement 'die on error' and 'reject flow' for output component

### PR DESCRIPTION
**Pull request by iistomin-gl**
**What is the current behavior?** (You can also link to an open issue here)
* NetSuite output component does not provide 'Die on error' option
* NetSuite output component does not provide 'Success/Reject' outgoing flows

**What is the new behavior?**
* NetSuite output component provides 'Die on error' option
* NetSuite output component provides 'Success/Reject' outgoing flows

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [X] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications:
* Implemented migration on properties post deserialization stage

**Other information**:
JIRA issue: https://jira.talendforge.org/browse/TDI-38601

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/170/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/170/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/171/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/171/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/172/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/172/)

File: components/components-netsuite/netsuite-definition/pom.xml [hunk 1]
```diff
@@ -32,6 +32,11 @@
             <artifactId>components-api</artifactId>
         </dependency>
 
+        <!--<dependency>-->
```
Comment (gonchar-ivan) [inline components/components-netsuite/netsuite-definition/pom.xml]: please, check this commented dependency. Maybe you want to remove it?

Comment (iistomin-gl) [inline components/components-netsuite/netsuite-definition/pom.xml]: Ok

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/190/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/190/)

**Pull request merged**